src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/gcs/GcsAuthorizationProvider.scala

```diff
@@ -0,0 +1,19 @@
+
+package org.broadinstitute.dsde.agora.server.dataaccess.gcs
+
+import org.broadinstitute.dsde.agora.server.business.AgoraPermissions._
+import org.broadinstitute.dsde.agora.server.business.{AuthorizationProvider, AgoraAcl, AuthorizedAgoraEntity}
+import org.broadinstitute.dsde.agora.server.model.AgoraEntity
+
+object GcsAuthorizationProvider extends AuthorizationProvider {
+  override def authorizationsForEntity(agoraEntity: AgoraEntity): AuthorizedAgoraEntity = {
+    //TODO call google here and do GACL -> AgAcl translation
```

Comment:
You are going to need to pass to Google the common-name of the user whose permissions you want to check. This is not the owner of the method, but the user making the request. So we will need to add the usernameFromCookie() directive at AgoraService:95 to the queryRoute and queryByNamespaceNameSnapshotIdRoute, and pass this down into here.